fairlearn/metrics/_metric_frame.py

```diff
@@ -0,0 +1,515 @@
+        """
+        return self._cf_names
+
+    @property
+    def sensitive_levels(self) -> List[str]:
+        """Return a list of the feature names which are produced by sensitive features.
+
+        In cases where the ``by_group`` property has a :class:`pd.MultiIndex`
+        index, this identifies which elements of the index are sensitive features.
+
+        Returns
+        -------
+        List[str]
+            List of names, which can be used in calls to
+            :meth:`pandas.DataFrame.groupby` etc.
+        """
+        return self._sf_names
+
+    def group_max(self) -> Union[pd.Series, pd.DataFrame]:
+        """Return the maximum value of the metric over the sensitive features.
+
+        This method computes the maximum value over all combinations of
+        sensitive features for each underlying metric function in the ``by_groups``
+        property (it will only succeed if all the underlying metric
+        functions return scalar values). If there are no control
+        features, then the result will be a :class:`pandas.Series`,
+        with rows corresponding to the underying metrics.
+
+        Returns
+        -------
+        pandas.Series or pandas.DataFrame
+            If there are control features, then the result is a
+            :class:`pandas.DataFrame` with columns corresponding to the
+            underlying metrics, and rows indexed by the classes identified
+            by the control features. Otherwise, a :class:`pandas.Series`
+            indexed by the names of the underlying metrics.
+        """
+        if not self.control_levels:
+            result = pd.Series(index=self.by_group.columns, dtype='object')
+            for m in result.index:
+                max_val = self.by_group[m].max()
+                result[m] = max_val
+        else:
+            result = self.by_group.groupby(level=self.control_levels).max()
+        return result
+
+    def group_min(self) -> Union[pd.Series, pd.DataFrame]:
+        """Return the minimum value of the metric over the sensitive features.
+
+        This method computes the minimum value over all combinations of
+        sensitive features for each underlying metric function in the ``by_groups``
+        property (it will only succeed if all the underlying metric
+        functions return scalar values). If there are no control
+        features, then the result will be a :class:`pandas.Series`,
+        with rows corresponding to the underying metrics.
+
+        Returns
+        -------
+        pandas.Series or pandas.DataFrame
+            If there are control features, then the result is a
+            :class:`pandas.DataFrame` with columns corresponding to the
+            underlying metrics, and rows indexed by the classes identified
+            by the control features. Otherwise, a :class:`pandas.Series`
+            indexed by the names of the underlying metrics.
+        """
+        if not self.control_levels:
```

Comment:
simplify to one line

Reply:
Some simplifications done, but internals can wait on a refactor